### PR TITLE
nchat: 3.67 → 4.86

### LINKED ISSUE #318085: Update request: nchat 3.67 → 4.86
- Package name: nchat
- Latest released version: 4.86

- Current version on the unstable channel: 3.67
- Current version on the stable/release channel: 3.67

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

The only PR open is a bot update for an outdated version that was never merged. 

**Notify maintainers**

@sikmir 

-----

Note for maintainers: Please tag this issue in your PR.

I've gotten the newest package to build with this config and command:

```nix
{ lib, stdenv,  fetchFromGitHub, cmake, gperf
, file, ncurses, openssl, readline, sqlite, zlib
, AppKit ? null, Cocoa ? null, Foundation ? null
}:

stdenv.mkDerivation rec {
  pname = "nchat";
  version = "4.86";

  src = fetchFromGitHub {
    owner = "d99kris";
    repo = "nchat";
    rev = "v${version}";
    hash = "sha256-ju9FabCxh/OvEFZee6ZzJEG5RpaCwR297SFk7leKnhA=";
  };

  postPatch = ''
    substituteInPlace lib/tgchat/ext/td/CMakeLists.txt \
      --replace "get_git_head_revision" "#get_git_head_revision"
  '';

  nativeBuildInputs = [ cmake gperf ];

  buildInputs = [
    file # for libmagic
    ncurses
    openssl
    readline
    sqlite
    zlib
  ] ++ lib.optionals stdenv.isDarwin [ AppKit Cocoa Foundation ];

  cmakeFlags = [
    "-DHAS_WHATSAPP=OFF" # go module build required
  ];

  meta = with lib; {
    description = "Terminal-based chat client with support for Telegram and WhatsApp";
    mainProgram = "nchat";
    homepage = "https://github.com/d99kris/nchat";
    license = licenses.mit;
    maintainers = with maintainers; [ sikmir ];
    platforms = platforms.unix;
  };
}
```
```sh
$ nix-build -E "with import <nixpkgs> {}; callPackage ./default.nix {}"
```

I don't know nix yet. I'm just hacking my way around

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by luftmensch-luftmensch**
## Description of changes

Closes #318085

Updated also location of the program

+ [Release notes](https://github.com/d99kris/nchat/releases/tag/v4.86)

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/nc/nchat/package.nix [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      openssl
+      readline
+      sqlite
+      zlib
+    ]
+    ++ lib.optionals stdenv.isDarwin [
+      AppKit
+      Cocoa
+      Foundation
+    ];
```
Comment (isabelroses) [inline pkgs/by-name/nc/nchat/package.nix:43]:
```suggestion
    ++ lib.optionals stdenv.isDarwin (
      with darwin.apple_sdk.frameworks; [
        AppKit
        Cocoa
        Foundation
      ]
    );
```

Comment (luftmensch-luftmensch) [reply pkgs/by-name/nc/nchat/package.nix]: @isabelroses  Applied suggestion!

**Review by isabelroses — APPROVED** [review]

**Comment by sikmir** [comment]
```
In file included from /tmp/nix-build-nchat-4.86.drv-0/source/lib/tgchat/src/tgchat.cpp:20:
/tmp/nix-build-nchat-4.86.drv-0/source/lib/tgchat/ext/td/td/telegram/Client.h:11:10: fatal error: 'td/telegram/td_api.h' file not found
#include "td/telegram/td_api.h"
         ^~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

**Comment by luftmensch-luftmensch** [comment]
> ```
> In file included from /tmp/nix-build-nchat-4.86.drv-0/source/lib/tgchat/src/tgchat.cpp:20:
> /tmp/nix-build-nchat-4.86.drv-0/source/lib/tgchat/ext/td/td/telegram/Client.h:11:10: fatal error: 'td/telegram/td_api.h' file not found
> #include "td/telegram/td_api.h"
>          ^~~~~~~~~~~~~~~~~~~~~~
> 1 error generated.
> ```

How did you get it? I'm unable to reproduce it

**Comment by sikmir** [comment]
> How did you get it? I'm unable to reproduce it

https://github.com/NixOS/nixpkgs/pull/318096/checks?check_run_id=25991476547

**Comment by Steinhagen** [comment]
Is there any update for this?

**Comment by Steinhagen** [comment]
To speed things a bit, I've cherry-picked your commit, changed to the latest version and created a new PR here: https://github.com/NixOS/nixpkgs/pull/345606

**Comment by Steinhagen** [comment]
The build was updated and fixed in the previous mentioned PR.

**Pull request closed without merging**